deploy/gcp/main.tf

```diff
@@ -0,0 +1,290 @@
+
+  node_config {
+    machine_type = "${var.tikv_instance_type}"
+    local_ssd_count = 1
+    taint {
+      effect = "NO_SCHEDULE"
+      key = "dedicated"
+      value = "tikv"
+    }
+    labels {
+      dedicated = "tikv"
+    }
+    tags = ["tikv"]
+    oauth_scopes = ["storage-ro", "logging-write", "monitoring"]
+
+  }
+
+}
+
+resource "google_container_node_pool" "tidb_pool" {
+  provider = "google-beta"
+  project = "${var.GCP_PROJECT}"
+  cluster = "${google_container_cluster.cluster.name}"
+  location = "${google_container_cluster.cluster.location}"
+  name = "tidb-pool"
+  initial_node_count = "${var.tidb_count}"
+
+  node_config {
+    machine_type = "${var.tidb_instance_type}"
+    taint {
+      effect = "NO_SCHEDULE"
+      key = "dedicated"
+      value = "tidb"
+    }
+    labels {
+      dedicated = "tidb"
+    }
+    tags = ["tidb"]
+    oauth_scopes = ["storage-ro", "logging-write", "monitoring"]
+  }
+
+}
+
+resource "google_container_node_pool" "monitor_pool" {
+  project = "${var.GCP_PROJECT}"
+  cluster = "${google_container_cluster.cluster.name}"
+  location = "${google_container_cluster.cluster.location}"
+  name = "monitor-pool"
+  initial_node_count = "1"
+
+  node_config {
+    machine_type = "${var.monitor_instance_type}"
+    tags = ["monitor"]
+    oauth_scopes = ["storage-ro", "logging-write", "monitoring"]
+  }
+
+}
+
+resource "google_compute_firewall" "allow_ssh_bastion" {
+  name = "allow-ssh-bastion"
+  network = "${google_compute_network.vpc_network.self_link}"
+  project = "${var.GCP_PROJECT}"
+
+  allow {
+    protocol = "tcp"
+    ports = ["22"]
+  }
+  source_ranges = ["0.0.0.0/0"]
+  target_tags = ["bastion"]
+}
+
+resource "google_compute_firewall" "allow_mysql_from_bastion" {
+  name = "allow-mysql-from-bastion"
+  network = "${google_compute_network.vpc_network.self_link}"
+  project = "${var.GCP_PROJECT}"
+
+  allow {
+    protocol = "tcp"
+    ports = ["4000"]
+  }
+  source_tags = ["bastion"]
+  target_tags = ["tidb"]
+}
+
+resource "google_compute_firewall" "allow_ssh_from_bastion" {
+  name = "allow-ssh-from-bastion"
+  network = "${google_compute_network.vpc_network.self_link}"
+  project = "${var.GCP_PROJECT}"
+
+  allow {
+    protocol = "tcp"
+    ports = ["22"]
+  }
+  source_tags = ["bastion"]
+  target_tags = ["tidb", "tikv", "pd", "monitor"]
+}
+
+resource "google_compute_instance" "bastion" {
+  project = "${var.GCP_PROJECT}"
+  zone = "${var.GCP_REGION}-a"
+  machine_type = "${var.bastion_instance_type}"
+  name = "bastion"
+  "boot_disk" {
+    initialize_params {
+      image = "ubuntu-os-cloud/ubuntu-1804-lts"
+    }
+  }
+  "network_interface" {
+    subnetwork = "${google_compute_subnetwork.public_subnet.self_link}"
+    access_config {}
+  }
+  tags = ["bastion"]
+
+  metadata_startup_script = "sudo apt-get install -y mysql-client && curl -s https://packagecloud.io/install/repositories/akopytov/sysbench/script.rpm.sh | bash && sudo apt-get -y install sysbench"
```

Comment:
Ubuntu image should use `script.deb.sh`. See here https://github.com/akopytov/sysbench#linux